### PR TITLE
Variables: replaces UUID with name for a more performant lookup in TemplateSrv

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
This PR will change so variables uses the name field as the id. This way when we lookup a variable in TemplateSrv we need not loop through an array and instead can access the variable directly. This has the unfortunate side effect that we need to sync the id field with the name field.

**Which issue(s) this PR fixes**:
Relates to #22564

**Special notes for your reviewer**:
Renaming new variables and existing variables should be a good place to start looking for bugs :)

### CODE REVIEW

**Review by dprokop — APPROVED** [review]
Looks good!

**Comment by mckn** [comment]
Could we not just kill the id and use the name instead? Sorry, for the stupid question but I think I have missed something 🙈  @hugohaggmark

**Comment by hugohaggmark** [comment]
> Could we not just kill the id and use the name instead? Sorry, for the stupid question but I think I have missed something 🙈 @hugohaggmark

Yes, we can probably do that when we implement #20703. But for now when we rename a "new variable" we get into trouble if we use `name` as `id`.  Hard to explain in text, we can have a chat about it.

File: public/app/features/variables/interval/actions.test.ts [hunk 1]
```diff
@@ -91,9 +89,8 @@ describe('interval actions', () => {
   describe('when updateAutoValue is dispatched', () => {
     describe('and auto is false', () => {
       it('then no dependencies are called', async () => {
-        const interval = variableBuilder
-          .interval()
-          .withUUID('0')
+        const interval = intervalBuilder()
```
Comment (mckn) [inline public/app/features/variables/interval/actions.test.ts]: Reads a lot better! Nice! 🙌

**Review by mckn — APPROVED** [review]
Looks really good! Awesome job @hugohaggmark 

proud

**Pull request merged**